board/app/controllers/thread_controller.php

```diff
@@ -1,107 +1,145 @@
+                    
+        switch ($current_page) { 
+            case self::CREATE_THREAD:
+                break;
+
+            case self::CREATE_THREAD_END:
+                $thread->title = Param::get('title'); 
+                $thread->category = Param::get('category');
+                $thread->user_id = $_SESSION['user_id'];
+                $comment->body = Param::get('body');
+                $comment->user_id = $_SESSION['user_id'];
+                $comment->username = $_SESSION['username'];
+          
+                try {
+                    $thread->create($comment);
+                } catch (ValidationException $e) {
+                    $current_page = self::CREATE_THREAD;
+                }
+                break;
+                
             default:
                 throw new NotFoundException("{$current_page} is not found");
-            break;
-        }  
+                break;
+            }  
 
         $this->set(get_defined_vars());
         $this->render($current_page);
     }
- 
-    //Displays maximum number of threads per page
-    public function index() 
+              
+    /*
+    *Displays the comments of the thread
+    */
+    public function view() 
     {
-        $per_page = self::MAX_THREAD_PER_PAGE; 
-        $current_page = Param::get('page', 1);
-        $pagination = new SimplePagination($current_page, $per_page) ;
+        $per_page = MAX_DATA_PER_PAGE;
+        $current_page = Param::get(PAGE, 1);
+        $pagination = new SimplePagination($current_page, $per_page);      
 
-        $threads = Thread::getAll($pagination->start_index -1, $pagination->count + 1);
-        $pagination->checkLastPage($threads);
+        $thread_id = Param::get('thread_id'); 
+        $thread = Thread::get($thread_id); 
+        $comments = Comment::getAll($pagination->start_index -1, $pagination->count + 1, $thread_id);
+       
+        $total = Comment::countAll($thread_id);
+        $pagination->checkLastPage($comments);
+        $pages = ceil($total / $per_page);
+        $this->set(get_defined_vars());
+    }
+
+    public function delete_thread() 
+    { 
+        $thread = Thread::get(Param::get('thread_id'));
+        $thread->deleteThread($_SESSION['user_id']);
+        $this->set(get_defined_vars());
+        $this->render('thread/delete_thread');
+    }  
+
+    /*
+    * Sorting of threads
+    */
 
+    public function index() 
+    {
+        $per_page = MAX_DATA_PER_PAGE;
+        $current_page = Param::get(PAGE, 1);
+        $pagination = new SimplePagination($current_page, $per_page); 
+        $threads = Thread::getAll($pagination->start_index -1, $pagination->count + 1);   
+        $pagination->checkLastPage($threads);
         $total = Thread::CountAll();
         $pages = ceil($total / $per_page);
-    
         $this->set(get_defined_vars()); 
-    }   
+    }  
 
-    //Displays the comments of the thread
-    public function view() 
+    public function my_threads()
```

Comment:
index() po displays all the threads.
my_threads po displays my own threads.

changed my_threads() po to own_threads().
Thank you :bow:
--> https://github.com/maragatchalian/FoodFacts/commit/cc8cbd400062790516056a3282b753821f126d76